tests/gradients/test_gradient_transform.py

```diff
@@ -0,0 +1,403 @@
+        res = grad_fn(w)
+
+        x, y = w
+        expected = np.array([[-np.sin(x), 0], [0, -2 * np.cos(y) * np.sin(y)]])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_decorator(self, tol):
+        """Test that a gradient transform decorating a QNode
+        acts correctly"""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.gradients.param_shift
+        @qml.qnode(dev)
+        def circuit(weights):
+            qml.RX(weights[0], wires=[0])
+            qml.RY(weights[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliX(1))
+
+        w = np.array([0.543, -0.654], requires_grad=True)
+        res = circuit(w)
+
+        x, y = w
+        expected = np.array([[-np.sin(x), 0], [0, -2 * np.cos(y) * np.sin(y)]])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_passing_arguments(self, mocker, tol):
+        """Test that a gradient transform correctly
+        passes arguments"""
+        dev = qml.device("default.qubit", wires=2)
+        spy = mocker.spy(qml.gradients.parameter_shift, "expval_param_shift")
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            qml.RX(weights[0], wires=[0])
+            qml.RY(weights[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliX(1))
+
+        grad_fn = qml.gradients.param_shift(circuit, shift=np.pi / 4)
+
+        w = np.array([0.543, -0.654], requires_grad=True)
+        res = grad_fn(w)
+
+        x, y = w
+        expected = np.array([[-np.sin(x), 0], [0, -2 * np.cos(y) * np.sin(y)]])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        assert spy.call_args[0][2] == np.pi / 4
+
+    def test_expansion(self, mocker, tol):
+        """Test that a gradient transform correctly
+        expands gates with no gradient recipe"""
+        dev = qml.device("default.qubit", wires=2)
+        spy = mocker.spy(qml.gradients.parameter_shift, "expval_param_shift")
+
+        class NonDiffRXGate(qml.PhaseShift):
+            grad_method = "F"
+
+            @staticmethod
+            def decomposition(x, wires):
+                return [qml.RX(x, wires=wires)]
+
+        @qml.qnode(dev)
+        def circuit(weights):
+            NonDiffRXGate(weights[0], wires=[0])
+            qml.RY(weights[1], wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliX(1))
+
+        grad_fn = qml.gradients.param_shift(circuit)
+
+        w = np.array([0.543, -0.654], requires_grad=True)
+        res = grad_fn(w)
+
+        x, y = w
+        expected = np.array([[-np.sin(x), 0], [0, -2 * np.cos(y) * np.sin(y)]])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+        assert spy.call_args[0][0].operations[0].name == "RX"
+
+    def test_permutated_arguments(self, tol):

```

Comment:
```suggestion
    def test_permuted_arguments(self, tol):
```

Reply:
huh, this must be a case of [british english vs. canadian english](https://www.google.com/search?client=firefox-b-d&q=permutated)? It reminds me of 'oriented' vs 'orientated' 😆